curriculum/challenges/english/07-scientific-computing-with-python/learn-the-bisection-method-by-finding-the-square-root-of-a-number/65ef198fde24dfb7ff675b42.md

```diff
@@ -0,0 +1,46 @@
+---
+id: 65ef198fde24dfb7ff675b42
+title: Step 4
+challengeType: 20
+dashedName: step-4
+---
+
+# --description--
+
+If the `square_target` is less than `0`, raise a `ValueError` with the message `'Square root of negative number is not defined in real numbers'`. Don't forget to remove the `pass` keyword.
```

Comment:
Reminder: `valueError` is introduced in the refactored Sudoku solver, might need to update the instruction based on the final order of this project.

Reply:
Now the `ValueError` is mentioned in the cipher project, too.